include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -979,6 +984,319 @@ namespace internal
+       * cell from the vector we actually store.
+       */
+      virtual void
+      get_function_gradients(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<Tensor<1, DoFHandlerType::space_dimension>>
+          & /*patch_gradients*/) const override
+      {}
+
+      /**
+       * Given a FEValuesBase object, extract the gradients on the present
+       * cell from the vector we actually store. This function does the same
+       * as the one above but for vector-valued finite elements.
+       */
+      virtual void
+      get_function_gradients(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<std::vector<Tensor<1, DoFHandlerType::space_dimension>>>
+          & /*patch_gradients_system*/) const override
+      {}
+
+      /**
+       * Given a FEValuesBase object, extract the second derivatives on the
+       * present cell from the vector we actually store.
+       */
+      virtual void
+      get_function_hessians(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<Tensor<2, DoFHandlerType::space_dimension>>
+          & /*patch_hessians*/) const override
+      {}
+
+      /**
+       * Given a FEValuesBase object, extract the second derivatives on the
+       * present cell from the vector we actually store. This function does
+       * the same as the one above but for vector-valued finite elements.
+       */
+      virtual void
+      get_function_hessians(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<std::vector<Tensor<2, DoFHandlerType::space_dimension>>>
+          & /*patch_hessians_system*/) const override
+      {}
+
+      /**
+       * Return whether the data represented by (a derived class of) this object
+       * represents a complex-valued (as opposed to real-valued) information.
+       */
+      virtual bool
+      is_complex_valued() const override
+      {
+        return false;
+      }
+
+      /**
+       * Clear all references to the vectors.
+       */
+      virtual void
+      clear() override
+      {
+        vectors = nullptr;
+      }
+
+      /**
+       * Determine an estimate for the memory consumption (in bytes) of this
+       * object.
+       */
+      virtual std::size_t
+      memory_consumption() const override
+      {
+        return sizeof(vectors);
```

Comment:
Why are you not using `MemoryConsumption::memory_consumption(vectors)`?

Reply:
This is just a pointer, so I think this is correct as is.

Reply:
The question is who owns the vectors? But I think that the user owns the object, right?

Reply:
Yes, this is passed down from ``add_mg_data_vector``. The data belongs to the user needs to survive generating the output (same way as any other ``add_data_vector function``)